### ISSUE TITLE
lodash/collection-return false positive

### ISSUE THREAD

**Issue by mik01aj** [wix/eslint-plugin-lodash#75]
In this code example:

```
var promises = _.map(this.state.value, issueId =>
    adminApiClient.addNomination(issueId).then(() => {
        console.log('promise done!');
    })
);
```

I get this warning:

```
Do not use _.map without returning a value     lodash/collection-return
```

But this is not true, because the `_.map` callback returns a promise. 🐞 

Note that when I remove the `then` part, it is OK again.

**Comment by ganimomer** [comment]
Hi,
Sorry for the late reply. Looks like a bug with our handling arrow functions. on it.
